### DISCUSSION TITLE
Email address on login interface is case-sensitive

### DISCUSSION THREAD

**Discussion by pc-erin** [keystonejs/keystone#8963] [Questions] [answered]
As far as I know, email addresses are generally not case-sensitive, and some users tend to type their email with capital letters in them.
As a result, people end up trying to log in and not knowing why it doesn't work. Making the email comparison case-insensitive would be helpful.

**Comment by dcousens** [comment]
You can opt for your `email` field to be case-insensitive in your Keystone schema by using a [`resolveInput`](https://keystonejs.com/docs/config/hooks#resolve-input) hook.

*edit*: this won't work for filters

**Reply by pc-erin** [reply]
I don't see an option for case-sensitivity in the text field. Do you mean use a resolveInput hook to coerce everything to lowercase?

**Reply by dcousens** [reply]
@pc-erin interestingly, yes, I thought we had a property on the `text` field, but you're right, we don't.
A `resolveInput` field hook would probably be the easiest approach, sorry for the confusion.

**Reply by pc-erin** [reply]
@dcousens I just tried that. It works in the panel (like when modifying a user's email), but the hook doesn't fire when the user logs in. Which kind of makes sense since hooks only fire for create, update, and delete operations. It wouldn't have anything to do during a query op.

**Reply by dcousens** [reply]
@pc-erin indeed :disappointed: 

We can probably temporarily resolve this by adding an option to `@keystone-6/auth` that changes the search filters to use `is_i` for queries. Alternatively, I might re-think some of these hooks.

**Comment by pc-erin** [comment]
My workaround for this so far is to add the following as `db.extendPrismaSchema`:

```ts
extendPrismaSchema(schema) {
  return schema
	  .split('\n')
	  .map(line => {
		  if (line.includes('email') || line.includes('clientEmail')) {
			  return line + ' @postgresql.Citext'
		  }
		  return line
	  })
	  .join('\n')
 },
```

Where email and clientEmail are the fields I wanted to change to case insensitive.
You also need to be using postgres and enable the citext type with `CREATE EXTENSION "citext";` on the relevant database.

**Reply by dcousens** [reply, marked as answer]
An alternative to `extendPrismaSchema` is to use `db.nativeType: 'citext'` for the `text` field type: https://github.com/keystonejs/keystone/blob/1ed8e2ef297e38c1fac4ac4a65792d644ff38bb9/packages/core/src/fields/types/text/index.ts#L49
